app/questionnaire/questionnaire_schema.py

```diff
@@ -166,8 +166,11 @@ def is_view_submitted_response_enabled(self) -> bool:
         is_enabled: bool = schema.get("view_response", False)
         return is_enabled
 
-    def _get_sections_by_id(self) -> dict[str, ImmutableDict]:
-        return {section["id"]: section for section in self.json.get("sections", [])}
+    def _get_sections_by_id(self) -> dict[str, ImmutableDict[str, Any]]:
+        return {
+            section["id"]: section
+            for section in self.json.get("sections", ImmutableDict({}))
```

Comment:
Yep, hundreds of test failures, looks like we have a lot of tests that use QuestionnaireSchema({}) or initialise it with a single key specific to that test, but since its always mandatory I can go through and update these with an empty sections dict so its closer to an actual run

Reply:
I think leave it as it is for now, when we come to add actual data models (data classes for the schemas), we will have to handle it then.